File: performance/performance01.jl

function performance01()
  N=1000000
  M=10

  tic()

  for l=1:10
    x=zeros(Complex128, N)
    for i=0:M-1
      x += 1
    end

    for i=0:(N-1)
      x[i+1] += i
    end

    for i=0:M-1
      x += i*exp(2*pi*im*i*linspace(0,1,N))
    end

    for i=0:M-1
      x += i
    end
  end

  toc()
end

performance01()

File: performance/performance01.py
from numpy import *
import timeit

def performance01():
  N=1000000
  M=10

  for l in xrange(0,10):
    x=zeros((N),dtype=cdouble)
    for i in xrange(0,M):
      x += ones((N))

    for i in xrange(0,N):
      x[i] += i

    for i in xrange(0,M):
      x += i*exp(2*pi*1j*i*linspace(0,1,N))

    for i in xrange(0,M):
      x += i


tic = timeit.default_timer()
performance01()
toc = timeit.default_timer()
print(toc-tic)
